projectmanagement/src/main/java/life/qbic/projectmanagement/application/ProjectInformationService.java

```diff
@@ -58,6 +69,136 @@ public List<ProjectPreview> queryPreview(String filter, int offset, int limit,
     return new ArrayList<>(previewList);
   }
 
+  private Optional<Experiment> loadActiveExperimentForProject(Project project) {
+    return project.activeExperiment()
+        .map(experimentId ->
+            experimentRepository.find(experimentId)
+                .orElseThrow(
+                    () -> new RuntimeException("The active experiment does not exist anymore.")
+                    // should never happen; indicates dirty removal of experiment from db
+                ));
+  }
+
+  /**
+   * Adds species to the active experiment of a project. If no experiment is active, a new
+   * experiment is created and set as the active experiment.
+   *
+   * @param projectId the project for which to add the species
+   * @param species   the species to add
+   * @see Experiment#addSpecies(Collection)
+   */
+  public void addSpeciesToActiveExperiment(String projectId, Species... species) {
+    if (species.length < 1) {
+      return;
+    }
+    Arrays.stream(species).forEach(Objects::requireNonNull);
+
+    ProjectId id = ProjectId.parse(projectId);
+    Optional<Project> optionalProject = projectRepository.find(id);
+
+    optionalProject.ifPresentOrElse(
+        project -> {
+          loadActiveExperimentForProject(project)
+              .ifPresentOrElse(
+                  activeExperiment -> {
+                    activeExperiment.addSpecies(List.of(species));
+                    experimentRepository.update(activeExperiment);
+                  },
+                  () -> {
+                    Experiment experiment = Experiment.createForProject(project, List.of(),
+                        List.of(), List.of(species));
+                    experimentRepository.add(experiment);
+                    project.linkExperiment(experiment.experimentId());
+                    projectRepository.update(project);
```

Comment:
To increase readability, please reduce the consumer / function to own methods.

Reply:
As we decided to take the automatic creation of experimental variables out for the time being, this code vanished 😄